pkg/registry/events/event/strategy.go

```diff
@@ -85,8 +85,21 @@ func (eventStrategy) AllowUnconditionalUpdate() bool {
 }
 
 // SelectableFields returns a field set that represents the object.
-func SelectableFields(pip *coreapi.Event) fields.Set {
-	return generic.ObjectMetaFieldsSet(&pip.ObjectMeta, true)
+func SelectableFields(event *coreapi.Event) fields.Set {
+	objectMetaFieldsSet := generic.ObjectMetaFieldsSet(&event.ObjectMeta, true)
+	specificFieldsSet := fields.Set{
+		"involvedObject.kind":            event.InvolvedObject.Kind,
+		"involvedObject.namespace":       event.InvolvedObject.Namespace,
+		"involvedObject.name":            event.InvolvedObject.Name,
+		"involvedObject.uid":             string(event.InvolvedObject.UID),
+		"involvedObject.apiVersion":      event.InvolvedObject.APIVersion,
+		"involvedObject.resourceVersion": event.InvolvedObject.ResourceVersion,
+		"involvedObject.fieldPath":       event.InvolvedObject.FieldPath,
+		"reason":                         event.Reason,
+		"source":                         event.ReportingController,
```

Comment:
this should use events/{v1beta1,v1} field names:

* instead of `involvedObject`, use `regarding`
* instead of `source`, use `reportingController`